pydoctor/templatewriter/pages/sidebar.py

```diff
@@ -166,7 +172,7 @@ def _children(self, inherited: bool = False) -> Optional[List[Documentable]]:
                 return sorted((o for o in util.list_inherited_members(self.ob) if o.isVisible),
                               key=util.objects_order)
             else:
-                return None
+                assert False, "Use inherited=True only with Class instances"
```

Comment:
I appreciate trying to keep the diff small, but assertions in the negative path like this are somewhat confusing to read; it seems like there's going to be a path through at first and then there isn't.  When reading the assertion in a traceback it's less clear what's being asserted. Instead, I'd do:

```python3
assert isinstance(self.ob, Class), "Use inherited=True only with Class instances"
return sorted((o for o in util.list_inherited_members(self.ob) if o.isVisible), key=util.objects_order)
```

Sorry that's not a suggestion, but it modifies lines outside the diff…